otx/algorithms/segmentation/adapters/mmseg/task.py

```diff
@@ -0,0 +1,607 @@
+            mm_dataset,
+            samples_per_gpu=cfg.data.get("samples_per_gpu", 1),
+            workers_per_gpu=cfg.data.get("workers_per_gpu", 0),
+            num_gpus=len(cfg.gpu_ids),
+            dist=cfg.distributed,
+            seed=cfg.get("seed", None),
+            persistent_workers=False,
+            shuffle=False,
+        )
+
+        # Target classes
+        if "task_adapt" in cfg:
+            target_classes = cfg.task_adapt.final
+            if len(target_classes) < 1:
+                raise KeyError(
+                    f"target_classes={target_classes} is empty check the metadata from model ckpt or recipe "
+                    "configuration"
+                )
+        else:
+            target_classes = mm_dataset.CLASSES
+
+        # Model
+        model = self.build_model(cfg, fp16=cfg.get("fp16", False))
+        model.CLASSES = target_classes
+        model.eval()
+        feature_model = model.model_s if self._train_type == TrainType.Semisupervised else model
+        model = build_data_parallel(model, cfg, distributed=False)
+
+        # InferenceProgressCallback (Time Monitor enable into Infer task)
+        time_monitor = None
+        if cfg.get("custom_hooks", None):
+            time_monitor = [hook.time_monitor for hook in cfg.custom_hooks if hook.type == "OTXProgressHook"]
+            time_monitor = time_monitor[0] if time_monitor else None
+        if time_monitor is not None:
+
+            # pylint: disable=unused-argument
+            def pre_hook(module, inp):
+                time_monitor.on_test_batch_begin(None, None)
+
+            def hook(module, inp, outp):
+                time_monitor.on_test_batch_end(None, None)
+
+            model.register_forward_pre_hook(pre_hook)
+            model.register_forward_hook(hook)
+
+        eval_predictions = []
+        feature_vectors = []
+
+        if not dump_features:
+            feature_vector_hook: Union[nullcontext, BaseRecordingForwardHook] = nullcontext()
+        else:
+            feature_vector_hook = FeatureVectorHook(feature_model)
+
+        with feature_vector_hook:
+            for data in dataloader:
+                with torch.no_grad():
+                    result = model(return_loss=False, output_logits=True, **data)
+                eval_predictions.append(result)
+                if isinstance(feature_vector_hook, nullcontext):
+                    feature_vectors = [None] * len(mm_dataset)
+                else:
+                    feature_vectors = feature_vector_hook.records
+
+        assert len(eval_predictions) == len(feature_vectors), (
+            "Number of elements should be the same, however, number of outputs are ",
+            f"{len(eval_predictions)} and {len(feature_vectors)}",
+        )
+
+        outputs = dict(
+            classes=target_classes,
+            eval_predictions=eval_predictions,
+            feature_vectors=feature_vectors,
+        )
+        return outputs
+
+    def explain(self):
```

Comment:
Doesn't segmentation have an explain function?

Reply:
As far as I know, yes. There are no test cases for segmentation explain.